rubin_sim/maf/plots/skyproj_plotters.py

```diff
@@ -0,0 +1,523 @@
+
+    Parameters
+    ----------
+    center_ra : `float`
+        R.A. of the center of the circle (deg.).
+    center_decl : `float`
+        Decl. of the center of the circle (deg.).
+    radius : float, optional
+        Radius of the circle (deg.), by default 90.0
+    start_bear : int, optional
+        Bearing (E. of N.) of the start of the circle (deg.), by default 0
+    end_bear : int, optional
+        Bearing (E. of N.) of the end of the circle (deg.), by default 360
+    step : int, optional
+        Spacing of the points along the circle (deg.), by default 1
+
+    Returns
+    -------
+    circle : `pandas.DataFrame`
+        DataFrame with points in the circle.
+    """
+    ras = []
+    decls = []
+
+    bearing_angles = np.arange(start_bear, end_bear + step, step) * u.deg
+    center_coords = SkyCoord(center_ra * u.deg, center_decl * u.deg)
+    circle_coords = center_coords.directional_offset_by(bearing_angles, radius * u.deg)
+    bearings = bearing_angles.value.tolist()
+    ras = circle_coords.ra.deg.tolist()
+    decls = circle_coords.dec.deg.tolist()
+
+    # de-normalize RA so there are no large jumps, which can
+    # confuse cartopy + matplotlib
+
+    previous_ra = ras[0]
+    for ra_index in range(1, len(ras)):
+        if ras[ra_index] - previous_ra > 180:
+            ras[ra_index] -= 360
+        if ras[ra_index] - previous_ra < -180:
+            ras[ra_index] += 360
+        previous_ra = ras[ra_index]
+
+    circle = pd.DataFrame(
+        {
+            "bearing": bearings,
+            "ra": ras,
+            "decl": decls,
+        }
+    ).set_index("bearing")
+
+    return circle
+
+
+class SkyprojPlotter(BasePlotter, abc.ABC):
+    """Base class for all plotters that use the skyproj module."""
+
+    # Store a mapping between figures and existing SkyprojPlotters, but use
+    # weak keys so that the figure can be cleaned up by the garbage collector,
+    # and when it is entry in the dictionary automatically deleted.
+    skyproj_instances = WeakKeyDictionary()
+
+    default_decorations = ["ecliptic", "galactic_plane"]
+
+    default_colorbar_kwargs = {
+        "location": "bottom",
+        "shrink": 0.75,
+        "aspect": 25,
+        "pad": 0.1,
+        "orientation": "horizontal",
+    }
+
+    ecliptic_kwargs = {"edgecolor": "green"}
+    galactic_plane_kwargs = {"edgecolor": "blue"}
+    sun_kwargs = {"color": "yellow"}
+    moon_kwargs = {"color": "orange"}
+    horizon_kwargs = {"edgecolor": "black", "linewidth": 3}
+
+    def __init__(self):
+        super().__init__()
+        # Customize our plotters members for our new plot
+
+        self.plot_type = "Skyproj"
+        self.default_plot_dict = {}
+        self.default_plot_dict.update(
+            {
+                "subplot": 111,
+                "skyproj": skyproj.MollweideSkyproj,
+                "skyproj_kwargs": {"lon_0": 0},
+                "decorations": self.default_decorations,
+            }
```

Comment:
I think there are some default kwargs that are expected to be present, for other plotters ... the BasePlotter suggests
```
        self.default_plot_dict = {
            "title": None,
            "xlabel": None,
            "label": None,
            "labelsize": None,
            "fontsize": None,
            "figsize": None,
        }
```

although, this is really only weakly required, and is mostly because when filling out the plotters, tests are not always done to ensure those keys are in the dictionaries. 
I see you tested for title before using it and same with title, so it's also not necessarily relevant in this case. 

I suppose making the list of plot dict items more standardized and more explained would be helpful. (I can't remember offhand, if "label", for instance, is used to make the color-bar label or if it's xlabel, for skymaps).

Reply:
I think it may be useful to add either a comment or to expand the default plot_dict to include the optional items -- 
"ModelObservatory", "labelsize", etc. 
Otherwise it's really hard to know what those keys are .. basically the default_plot_dict lets you have a place to look it up. 
And then you can just test if it's None rather than testing if it's present and if it's None.

Reply:
Okay, I've added these to the defaults, and added more description to the comment about what decorations need ModelObservatory and which don't.